AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/deploy/boundary/DeploymentBoundary.java

```diff
@@ -801,16 +801,19 @@ public String[] getLogFileNames(final int deploymentId) {
     }
 
     // TODO test
-    public String getDeploymentLog(String logName) throws IllegalAccessException {
+    public String getDeploymentLog(String logName) throws IOException {
         String logsPath = ConfigurationService.getProperty(ConfigKey.LOGS_PATH);
-        if (logName.contains(File.separator)) {
-            throw new IllegalAccessException("The log file contains a file separator (\"" + File.separator + "\"). For security reasons, this is not permitted!");
-        }
+
+        String name = logsPath + File.separator + logName;
+        File file = new File(name);
+
+        if (file.length() > 1_000_000) throw new IOException(String.format("%s is larger than 10 MB and cannot be shown.", file.getName()));
```

Comment:
Instead of an exception, could we instead read the first 10MB of the file and append an error to it?

Reply:
We could also try to stream the file instead to prevent loading it into memory completely.

Reply:
Both are good suggestions - I'll have to check it. I wonder if the users browser will crash if we stream large files.

Reply:
I've chosen a third option: 

Only the last 10mb of the log file is shown (if larger) and a message is prepended that says that some part of the log is missing...

I imagine that the interesting log entries are at the end of the log - and not at the beginning. 

@yvespp What do you think about it?